File: university-service/src/test/java/ua/com/foxminded/courseproject/repository/ScheduleRepositoryTest.java
package ua.com.foxminded.courseproject.repository;

import org.junit.jupiter.api.Test;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.boot.test.context.SpringBootTest;
import org.springframework.data.util.Streamable;
import org.springframework.test.context.jdbc.Sql;
import org.springframework.transaction.annotation.Transactional;
import ua.com.foxminded.courseproject.entity.Schedule;

import java.sql.Timestamp;
import java.util.List;
import java.util.UUID;

import static org.junit.jupiter.api.Assertions.assertEquals;

@SpringBootTest
@Sql(value = "classpath:initial_data.sql")
@Transactional
class ScheduleRepositoryTest {

    @Autowired
    private ScheduleRepository repository;

    @Test
    void findAll_shouldReturnListDays() {
        Integer expectedSize = 1;

        List<Schedule> schedules = Streamable.of(repository.findAll()).toList();

        assertEquals(expectedSize, schedules.size());
    }

    @Test
    void findById_shouldReturnDay_whenIdExists() {
        UUID id = UUID.fromString("b5fd8224-47ba-11ed-b878-0242ac120002");
        Timestamp expectedTime = Timestamp.valueOf("2022-09-01 08:00:00");

        Schedule actual = repository.findById(id).get();

        assertEquals(expectedTime, actual.getStartDate());
    }


}

File: university-service/src/test/java/ua/com/foxminded/courseproject/repository/ScheduleRepositoryTest.kt
//package ua.com.foxminded.courseproject.repository
//
//import org.junit.jupiter.api.Assertions
//import org.junit.jupiter.api.Test
//import org.springframework.beans.factory.annotation.Autowired
//import org.springframework.boot.test.autoconfigure.data.mongo.DataMongoTest
//import org.springframework.data.util.Streamable
//import ua.com.foxminded.courseproject.config.DBTestConfig
//import java.time.LocalDateTime
//import java.time.format.DateTimeFormatter
//import java.util.*
//
//@DataMongoTest
//internal open class ScheduleRepositoryTest : DBTestConfig() {
//    @Autowired
//    private lateinit var repository: ScheduleRepository
//
//    @Test
//    fun findAll_shouldReturnListDays() {
//        val expectedSize = 1
//
//        val schedules = Streamable.of(repository.findAll()).toList()
//
//        Assertions.assertEquals(expectedSize, schedules.size)
//    }
//
//    @Test
//    fun findById_shouldReturnDay_whenIdExists() {
//        val id = UUID.fromString("b5fd8224-47ba-11ed-b878-0242ac120002")
//        val expectedTime = LocalDateTime.parse("2022-09-01T08:00:00.000Z", DateTimeFormatter.ISO_DATE_TIME)
//
//        val actual = repository.findById(id).get()
//
//        Assertions.assertEquals(expectedTime, actual.startDate)
//    }
//}